components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/application-details.jsp

```diff
@@ -0,0 +1,123 @@
+<!--
+~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+~
+~ WSO2 Inc. licenses this file to you under the Apache License,
+~ Version 2.0 (the "License"); you may not use this file except
+~ in compliance with the License.
+~ You may obtain a copy of the License at
+~
+~    http://www.apache.org/licenses/LICENSE-2.0
+~
+~ Unless required by applicable law or agreed to in writing,
+~ software distributed under the License is distributed on an
+~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+~ KIND, either express or implied.  See the License for the
+~ specific language governing permissions and limitations
+~ under the License.
+-->
+
+<%@ page import="org.owasp.encoder.Encode"%>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="carbon" uri="http://wso2.org/projects/carbon/taglibs/carbontags.jar"%>
+<%@ page import="org.wso2.carbon.identity.application.mgt.ui.ApplicationBean" %>
+<%@ page import="org.wso2.carbon.identity.application.mgt.ui.util.ApplicationMgtUIUtil" %>
+
+<link href="css/idpmgt.css" rel="stylesheet" type="text/css" media="all"/>
+<carbon:breadcrumb label="breadcrumb.oauth.application" resourceBundle="org.wso2.carbon.identity.oauth.ui.i18n.Resources"
+                    topPage="true" request="<%=request%>" />
+<jsp:include page="../dialog/display_messages.jsp"/>
+
+<script type="text/javascript" src="../admin/js/main.js"></script>
+<script type="text/javascript" src="../identity/validation/js/identity-validate.js"></script>
+
+<%
+ApplicationBean appBean = ApplicationMgtUIUtil.getApplicationBeanFromSession(session, request.getParameter("spName"));
+if (appBean.getServiceProvider() == null || appBean.getServiceProvider().getApplicationName() == null) {
+// if appbean is not set properly redirect the user to list-service-provider.jsp.
+%>
+<script>
+location.href = "list-service-providers.jsp";
+</script>
+<%
+	return;
+}
+    String spName = appBean.getServiceProvider().getApplicationName();
+    String action = Encode.forJavaScript(request.getParameter("action"));
+    boolean isNeedToUpdate = false;
+
+    String attributeConsumingServiceIndex = request.getParameter("attrConServIndex");
+	if (attributeConsumingServiceIndex != null){
+		appBean.setAttributeConsumingServiceIndex(attributeConsumingServiceIndex);
+	}
+
+    String oauthapp = request.getParameter("oauthapp");
+    if (oauthapp!=null && "update".equals(action)){
+    	appBean.setOIDCAppName(oauthapp);
+    	isNeedToUpdate = true;
+    }
+
+    String oauthConsumerSecret = null;
+    if (session.getAttribute("oauth-consum-secret")!= null && ("update".equals(action) || "regenerate".equals(action))){
```

Comment:
@Biruntha Formatting issue. Please go through all the changes and fix formatting issues